lnd.go

```diff
@@ -78,6 +82,30 @@ func lndMain() error {
 	}
 	defer chanDB.Close()
 
+	// Only process macaroons if --no-macaroons isn't set.
+	var macaroonService *bakery.Service
+	if !cfg.NoMacaroons {
+		// Create the macaroon authentication/authorization service.
+		macaroonService, err = macaroons.NewService(cfg.DataDir)
+		if err != nil {
+			srvrLog.Errorf("unable to create macaroon service: %v",
+				err)
+			return err
+		}
+
+		// Create macaroon files for lncli to use if they don't exist.
+		if !fileExists(cfg.AdminMacPath) &&
```

Comment:
If the `genMacaroons` function writes the admin macaroon, but then `lnd` fails for some reason (crash, unplug, etc), we'll end up in a partial state that current isn't addressed. In this case, only the admin macaroon will have ben created meaning that the body of this conditional won't be executed. 

In order to address this the `&` should be changed to an `||`.